apps/api/src/app/workflows-v2/usecases/build-variable-schema/build-available-variable-schema.usecase.ts

```diff
@@ -40,18 +50,15 @@ export class BuildVariableSchemaUsecase {
               format: 'date-time',
               description: 'The last time the subscriber was online (optional)',
             },
-            data: {
-              type: 'object',
-              properties: {},
-              description: 'Additional data about the subscriber',
-              additionalProperties: true,
-            },
+            data: buildVariablesSchema(
+              subscriber && typeof subscriber === 'object' && 'data' in subscriber ? subscriber.data : {}
+            ),
```

Comment:
build the `subscriber.data` schema for variables based on the variables used in the control values liquid template